app/models/user.rb

```diff
@@ -7,4 +7,14 @@ class User < ApplicationRecord
   def self.exist?(id)
     find(id) rescue false
   end
+
+  def self.from_onmiauth(user_params)
+    user = find_or_create_by(uid: user_params[:uid])
+    user.email = user_params[:email]
+    user.provider = user_params[:provider]
+    user.token = user_params[:token]
+    user.refresh_token = user_params[:refresh_token]
+    user.save
+    user
+  end
```

Comment:
This is cool. So it creates or pulls up a user and basically updates their information from Google? I like it.